components/brave_rewards/resources/page/components/pageWallet.tsx

```diff
@@ -621,10 +621,24 @@ class PageWallet extends React.Component<Props, State> {
     return ''
   }
 
+  enableWalletProvider = (walletProvider: string) => {
+    const { currentCountryCode, parameters } = this.props.rewardsData
+    const { allow, block } = parameters.walletProviderRegions[walletProvider]
+
+    const empty = (array: string[]) => array.length === 0
+    if (empty(allow) === empty(block)) {
+      return false
+    }
+
+    const equals = (countryCode: string) => countryCode === currentCountryCode
+    return allow.some(equals) || !empty(block) && !block.some(equals)
```

Comment:
IIUC, the rule here is something like:

- If both `allow` and `block` are empty, then the provider is enabled.
- If both `allow` and `block` are non-empty, then the provider is enabled. (Is this a bad-data case?)
- If `allow` contains the provider, then it is enabled.
- If `block` is empty then it is enabled.
- If `block` does not contain the provider, then it is enabled.
- Otherwise it is not enabled.

Is that correct?

Reply:
No, that's not correct.
- If both `allow` and `block` are empty, then the provider is disabled. (bad-data case)
- If both `allow` and `block` are non-empty, then the provider is disabled. (bad-data case)
- If `allow` contains the country code, then the provider is enabled.
- If `allow` is empty, or doesn't contain the country code, and `block` is empty, then the provider is disabled.
- If `allow` is empty, or doesn't contain the country code, and `block` is non-empty, but contains the country code, then the provider is disabled.
- If `allow` is empty, or doesn't contain the country code, and `block` is non-empty, and doesn't contain the country code, then the provider is enabled.

Reply:
Changed to:
- If both `allow` and `block` are empty, then the provider is enabled.
- If `allow` contains the country code, then the provider is enabled.
- If `allow` doesn't contain the country code, and `block` is empty, then the provider is disabled.
- If `allow` doesn't contain the country code, and `block` is non-empty, but contains the country code, then the provider is disabled.
- If `allow` doesn't contain the country code, and `block` is non-empty, and doesn't contain the country code, then the provider is enabled.

as per our Slack conversation with @Miyayes and @zenparsing.